app/controllers/auth_controller.rb

```diff
@@ -26,10 +29,20 @@ def token
         token_type: 'Bearer',
         expires_in: time_to_live,
         scope: scope,
-        patient: Patient.all.first.id
+        patient: Patient.first&.id
       }
+    elsif params.key?(:code) && params.key?(:client_id)
+      render json: { error: 'invalid_client' }, status: :bad_request
     else
-      render_operation_outcome(code: 'unknown', http: :unauthorized, message: 'Unauthorized Code')
```

Comment:
Looks like Sam had the `token` endpoint return an operation outcome so we should probably keep that here instead of just rendering json.

Reply:
I asked him about that, there seems to a conflict between the OAuth spec and the FHIR spec, my implementations following the OAuth spec so he just said to keep it